### PR TITLE
En 5664 replace resolver finder with request handler

### PR DESCRIPTION
**Pull request by SebastianMarian**
* Replaced ResolverFinder with RequestHandler in sync package
* Added arguments to bootstrapper

### CODE REVIEW

File: process/sync/baseSync.go [hunk 1]
```diff
@@ -316,104 +313,72 @@ func (boot *baseBootstrap) cleanCachesAndStorageOnRollback(header data.HeaderHan
...
-			err := hdrRes.RequestDataFromNonce(currentNonce, boot.epochHandler.Epoch())
-			if err != nil {
-				log.Debug("RequestDataFromNonce", "error", err.Error())
-				continue
-			}
-
-			err = boot.requestedItemsHandler.Add(key)
-			if err != nil {
-				log.Trace("add requested item with error", "error", err.Error())
+			if boot.shardCoordinator.SelfId() == sharding.MetachainShardId {
```
Comment (miiu96) [inline process/sync/baseSync.go]: maybe you can change logic here to reduce complexity if haveHeader { continue}...

Comment (miiu96) [reply process/sync/baseSync.go]: and also add  one more method in _blockBootstrapper_ interface to can requestHeaderByNonce(nonde uint64) and also if instruction where is checked shard id can be removed.

File: process/sync/shardblock.go [hunk 2]
```diff
@@ -108,42 +66,13 @@ func NewShardBootstrap(
...
 	base.forkInfo = process.NewForkInfo()
 
-	miniBlocksRes, ok := miniBlocksResolver.(dataRetriever.MiniBlocksResolver)
-	if !ok {
-		return nil, process.ErrWrongTypeAssertion
-	}
-
-	boot.miniBlocksResolver = miniBlocksRes
-
 	boot.chRcvHdrNonce = make(chan bool)
```
Comment (miiu96) [inline process/sync/shardblock.go]: maybe you can create a method in baseSync that init common code
boot.chRcvHdrNonce = make(chan bool)
	boot.chRcvHdrHash = make(chan bool)
	boot.chRcvMiniBlocks = make(chan bool)
boot.chStopSync = make(chan bool)

	boot.setRequestedHeaderNonce(nil)
	boot.setRequestedHeaderHash(nil)
	boot.setRequestedMiniBlocks(nil)

File: process/sync/baseSync.go (hunk 1 above)
Comment (SebastianMarian) [reply process/sync/baseSync.go]: Done

File: process/sync/shardblock.go (hunk 2 above)
Comment (SebastianMarian) [reply process/sync/shardblock.go]: Done

**Pull request merged**